classes/controller/FrontController.php

```diff
@@ -1088,9 +1093,14 @@ public function registerJavascript($id, $relativePath, $params = array())
             'attributes' => null,
             'server' => 'local',
         ];
-
         $params = array_merge($default_params, $params);
 
+        if (Tools::hasMediaServer() && !Configuration::get('PS_JS_THEME_CACHE')) {
+            $addSlash = substr($relativePath, 0, 1) !== '/' ? '/' : '';
+            $themeDir = substr($id, 0, 6) == 'theme-' ? _THEMES_DIR_ . _THEME_NAME_: '';
+            $relativePath = Tools::getCurrentUrlProtocolPrefix() . Tools::getMediaServer($relativePath) . $addSlash . $themeDir . $relativePath;
+            $params['server'] = 'remote';
+        }
```

Comment:
It looks like this could be externalized in a function (ex: buildRelativePath), for readability and DRY purpose?

Reply:
Fixed